Framework/Crystal/src/SCDCalibratePanels.cpp

```diff
@@ -605,677 +153,384 @@ void SCDCalibratePanels::exec() {
+    Mantid::MantidVec &ColY = ColWksp->dataY(i);
+    Mantid::MantidVec &RowX = RowWksp->dataX(i);
+    Mantid::MantidVec &RowY = RowWksp->dataY(i);
+    Mantid::MantidVec &TofX = TofWksp->dataX(i);
+    Mantid::MantidVec &TofY = TofWksp->dataY(i);
+    int icount = 0;
+    for (int j = 0; j < nPeaks; j++) {
+      Peak peak = peaksWs->getPeak(j);
+      if (peak.getBankName() == bankName) {
+        try {
+          V3D q_lab =
+              (peak.getGoniometerMatrix() * UB) * peak.getHKL() * M_2_PI;
+          Peak theoretical(peak.getInstrument(), q_lab);
+          ColX[icount] = peak.getCol();
+          ColY[icount] = theoretical.getCol();
+          RowX[icount] = peak.getRow();
+          RowY[icount] = theoretical.getRow();
+          TofX[icount] = peak.getTOF();
+          TofY[icount] = theoretical.getTOF();
+        } catch (...) {
+          // g_log.debug() << "Problem only in printing peaks\n";
+        }
+        icount++;
+      }
+    }
+    PARALLEL_END_INTERUPT_REGION
   }
-  if (cell_type == ReducedCell::TRICLINIC() ||
-      cell_type == ReducedCell::MONOCLINIC()) {
-    fn->setParameter("Beta", beta);
+  PARALLEL_CHECK_INTERUPT_REGION
+
+  string colFilename = getProperty("ColFilename");
+  string rowFilename = getProperty("RowFilename");
+  string tofFilename = getProperty("TofFilename");
+  saveNexus(colFilename, ColWksp);
+  saveNexus(rowFilename, RowWksp);
+  saveNexus(tofFilename, TofWksp);
+}
+
+void SCDCalibratePanels::saveNexus(std::string outputFile,
+                                   MatrixWorkspace_sptr outputWS) {
+  IAlgorithm_sptr save = this->createChildAlgorithm("SaveNexus");
+  save->setProperty("InputWorkspace", outputWS);
+  save->setProperty("FileName", outputFile);
+  save->execute();
+}
+
+void SCDCalibratePanels::findL1(int nPeaks,
+                                DataObjects::PeaksWorkspace_sptr peaksWs) {
+  MatrixWorkspace_sptr L1WS = boost::dynamic_pointer_cast<MatrixWorkspace>(
+      API::WorkspaceFactory::Instance().create("Workspace2D", 1, 3 * nPeaks,
+                                               3 * nPeaks));
+
+  auto &outSp = L1WS->getSpectrum(0);
+  MantidVec &yVec = outSp.dataY();
+  MantidVec &eVec = outSp.dataE();
+  MantidVec &xVec = outSp.dataX();
+  std::fill(yVec.begin(), yVec.end(), 0.0);
+
+  for (int i = 0; i < nPeaks; i++) {
+    const DataObjects::Peak &peak = peaksWs->getPeak(i);
+
+    // 1/sigma is considered the weight for the fit
+    double weight = 1.;                // default is even weighting
+    if (peak.getSigmaIntensity() > 0.) // prefer weight by sigmaI
+      weight = 1.0 / peak.getSigmaIntensity();
+    else if (peak.getIntensity() > 0.) // next favorite weight by I
+      weight = 1.0 / peak.getIntensity();
+    else if (peak.getBinCount() > 0.) // then by counts in peak centre
+      weight = 1.0 / peak.getBinCount();
+    for (int j = 0; j < 3; j++) {
+      int k = i * 3 + j;
+      xVec[k] = i * 3 + j;
+      eVec[k] = weight;
+    }
```

Comment:
The unit tests pass if you replace this for loop with 
```cpp
xVec[i * 3] = i;
eVec[i * 3 + 2] = weight;
```

Looks like a copy/paste error in an earlier commit.